evm-contracts/src/v0.6/dev/AggregatorProxy.sol

```diff
@@ -125,7 +137,24 @@ contract AggregatorProxy is AggregatorInterface, AggregatorV3Interface, Owned {
       uint256 answeredInRound
     )
   {
-    return aggregator.getRoundData(_roundId);
+    uint16 reqEpoch;
+    uint256 reqRound;
+    (reqEpoch, reqRound) = parseRequestId(_requestId);
+    (
+      roundId,
+      answer,
+      startedAt,
+      updatedAt,
+      answeredInRound
+    ) = epochAggregators[reqEpoch].getRoundData(reqRound);
```

Comment:
after we pack epoch and aggregator into a struct (let's assume it's called `hotVars`), we can save some gas here:
```
if (reqEpoch == hotVars.epoch) {
    [...] = hotVars.aggregator.getRoundData(reqRound);
} else {
    [...] = epochAggregators[reqEpoch].getRoundData(reqRound);
}
```

(this assumes that get getRoundData will usually be called for the latest epoch)